### PR TITLE
activate PNK ux improvements

### PR DESCRIPTION
**Pull request by satello**
- autofill balance
- validate amount against max
- inform user if they already activated
- make arbitratorExtraData not required on Contract creation (random)
- validate eth address input

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "babel-polyfill": "^6.26.0",
     "blockies": "^0.0.2",
     "blockies-identicon": "^0.1.0",
+    "crypto-js": "^3.1.9-1",
```
Comment (n1c01a5) [inline package.json:13]: I think we can use web3 instead of this library. No?

File: src/Contracts/Form/index.js [hunk 2]
```diff
@@ -7,8 +7,15 @@ import Web3 from 'web3'
 import FontAwesome from 'react-fontawesome'
 import { deployContract } from '../../business/contract/action-creators'
 import Input from '../../Input'
+import SHA3 from 'crypto-js/sha3'
 import './Form.css'
 
+let sha3 = (value) => {
+  return SHA3(value, {
+    outputLength: 256
+  }).toString()
```
Comment (n1c01a5) [inline src/Contracts/Form/index.js:16]: maybe `web3.sha3(value, {encoding: 'hex'})`

File: package.json (hunk 1 above)
Comment (n1c01a5) [reply package.json:13]: or maybe use the library `ethers.js`, see https://docs.ethers.io/ethers.js/html/api-utils.html#cryptographic-functions

File: src/Contracts/Form/index.js [hunk 3]
```diff
@@ -122,6 +128,34 @@ const Form = props => {
...
+  address = address.replace('0x', '')
+  const addressHash = sha3(address.toLowerCase())
+  for (let i = 0; i < 40; i++) {
+    // the nth letter should be uppercase if the nth digit of casemap is 1
+    if ((parseInt(addressHash[i], 16) > 7 && address[i].toUpperCase() !== address[i]) || (parseInt(addressHash[i], 16) <= 7 && address[i].toLowerCase() !== address[i])) {
+      return false
+    }
+  }
+  return true
+}
```
Comment (n1c01a5) [inline src/Contracts/Form/index.js:157]: you can use `ethers.js` for that

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

**Review by n1c01a5 — CHANGES_REQUESTED** [review]
use `ethers.js`

File: package.json (hunk 1 above)
Comment (satello) [reply package.json:13]: no it is a different sha3 algorithm. I tried that one first. Pretty confusing that `web3.eth.sha3` is different than the Solidity `keccak256`. I can try the `ethers.js keccak256`. Should work

File: src/Contracts/Form/index.js (hunk 3 above)
Comment (satello) [reply src/Contracts/Form/index.js:157]: do you know which function offhand? I don't see it in the docs

File: src/Contracts/Form/index.js [hunk 4]
```diff
@@ -122,6 +128,34 @@ const Form = props => {
...
+    // check if it has the basic requirements of an address
+    return false
+  } else if (/^(0x)?[0-9a-f]{40}$/.test(address) || /^(0x)?[0-9A-F]{40}$/.test(address)) {
+    // If it's all small caps or all all caps, return true
+    return true
+  } else {
+    // Otherwise check each case
+    return isChecksumAddress(address)
+  }
+}
```
Comment (n1c01a5) [inline src/Contracts/Form/index.js:143]: I think there is a library for that else you can use this `const isAddr = addr => /^(0x)?[0-9a-f]{40}$/i.test(addr) || /^(0x)?[0-9a-f]{64}$/i.test(addr)`

File: src/Contracts/Form/index.js (hunk 3 above)
Comment (n1c01a5) [reply src/Contracts/Form/index.js:157]: yes you're right. Later we can put these functions in a `utils` folder

File: package.json (hunk 1 above)
Comment (n1c01a5) [reply package.json:13]: ok you can try `ethers.js`

**Review by n1c01a5 — APPROVED** [review]

**Pull request merged**